app/routes/admin/users/add.tsx

```diff
@@ -0,0 +1,175 @@
+import { useCallback } from 'react'
+
+import type { ActionArgs } from '@remix-run/node'
+import { json, redirect } from '@remix-run/node'
+import { useActionData, useTransition } from '@remix-run/react'
+
+import { Role } from '@prisma/client'
+
+import {
+  Button,
+  FormControl,
+  FormLabel,
+  Heading,
+  Icon,
+  Input,
+  Radio,
+  RadioGroup,
+  Text,
+} from '@chakra-ui/react'
+
+import { UserPlus } from 'lucide-react'
+import { useNavigate } from 'react-router'
+import { z } from 'zod'
+
+import Dialog from '~/components/atoms/Dialog'
+import DialogBody from '~/components/atoms/DialogBody'
+import DialogClose from '~/components/atoms/DialogClose'
+import DialogContent from '~/components/atoms/DialogContent'
+import DialogFooter from '~/components/atoms/DialogFooter'
+import DialogHeader from '~/components/atoms/DialogHeader'
+import DialogOverlay from '~/components/atoms/DialogOverlay'
+import Fieldset from '~/components/atoms/Fieldset'
+import Form from '~/components/atoms/Form'
+import PasswordInput from '~/components/atoms/PasswordInput'
+import ValidationMessages from '~/components/atoms/ValidationMessages'
+import { createUser } from '~/models/user.server'
+import { setMessage } from '~/session.server'
+import type { inferSafeParseErrors } from '~/utils'
+
+const UserFields = z.object({
```

Comment:
J'utilise ça comme convention pour mes schemas zod mais comme tu veux

```suggestion
const userAddSchema= z.object({})
type UserAddSchema = z.infer<typeof userAddSchema>
type UserAddSchemaErrors = inferSafeParseErrors<typeof UserAddSchema>
```